spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -184,13 +184,20 @@ public void restartJobExecution(long jobExecutionId) throws NoSuchJobExecutionEx
 		TaskDefinition taskDefinition = this.taskDefinitionRepository.findById(taskExecution.getTaskName())
 				.orElseThrow(() -> new NoSuchTaskDefinitionException(taskExecution.getTaskName()));
 
+		// Remove Duplicate Task Name from arg list if present.
+		List<String> updatedTaskArguments = new  ArrayList<>();
```

Comment:
Is there a reason why we add `spring.cloud.task.name` more than one time? Is this because the way Spring Cloud Task updates the arguments?

Reply:
@ilayaperumalg Great Question!    The TaskDefinition https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java#L118 adds it as a property during `@PostLoad`.  Now for Local and CF deployers that has no effect when we restart a batch application, because this property is not stored as a part of the execution.   However, with Kubernetes it transforms all properties to CommandLineArgs.   And thus the first launch of the task adds the property and this is transformed into a command line arg and stored as a part of the job params.   The restart then follows the same steps resulting in --spring.cloud.task.name=foo,foo.